scalding-core/src/main/java/com/twitter/scalding/cascading/DistCacheEnabledHadoopPlanner.java

```diff
@@ -0,0 +1,95 @@
+package com.twitter.scalding;
+
+import cascading.flow.FlowElement;
+import cascading.flow.planner.graph.ElementGraph;
+import cascading.flow.planner.graph.FlowElementGraph;
+import cascading.flow.planner.iso.transformer.ElementFactory;
+import cascading.flow.planner.rule.transformer.IntermediateTapElementFactory;
+import cascading.flow.planner.rule.RuleRegistry;
+import cascading.flow.hadoop.planner.HadoopPlanner;
+import cascading.pipe.Checkpoint;
+import cascading.pipe.Pipe;
+import cascading.property.ConfigDef;
+import cascading.tap.Tap;
+import cascading.tap.hadoop.DistCacheTap;
+import cascading.tap.hadoop.Hfs;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Same as HadoopPlanner, but with DistCacheTap enabled for rhs of HashJoins
+ *
+ * This is achieved by using a DistCacheTapElementFactory with associated
+ * cascading rules to match HashJoin expressions.
+ */
+public class DistCacheEnabledHadoopPlanner extends HadoopPlanner {
+
+  /* Key used for registering DistCacheTapElementFactory */
+  public static String DIST_CACHE_TAP = "cascading.registry.tap.distcache";
```

Comment:
Seems to be the convention for entries added to the `ruleRegistry` - https://github.com/cwensel/cascading/blob/wip-3.2/cascading-hadoop2-tez/src/main/java/cascading/flow/tez/planner/Hadoop2TezPlanner.java#L161. Those two also start with `cascading.registry.`